protocol/nats_jetstream/v2/receiver.go

```diff
@@ -38,6 +39,20 @@ func (r *Receiver) MsgHandler(msg *nats.Msg) {
 	r.incoming <- msgErr{msg: NewMessage(msg)}
 }
 
+// MsgHandlerV2 implements jetstream.MessageHandler and publishes messages onto our internal incoming channel to be delivered
+// via r.Receive(ctx)
+func (r *Receiver) MsgHandlerV2(msg jetstream.Msg) {
+	// *nats.Msg does not implement jetstream.Msg.
+	// jetstream.MessageHandler.Ack() returns error where nats.Msg.Ack() returns nothing
+	natsMessage := &nats.Msg{
+		Subject: msg.Subject(),
+		Reply:   msg.Reply(),
+		Header:  msg.Headers(),
+		Data:    msg.Data(),
+	}
+	r.incoming <- msgErr{msg: NewMessage(natsMessage)}
```

Comment:
Another option would be to define an interface that both `nats.Msg` and `jetstream.Msg` could be wrapped to implement, e.g.

```go
type natsData interface {
  Subject() string
  Reply() string
  Headers() nats.Header
  Data() []byte
}
```

But this seems likely to be just as effective.